### PR TITLE
[FIX] mail: fix click event handling when renaming DM

### PR DESCRIPTION
**Pull request by xdu-odoo**
task-2414811

### CODE REVIEW

**Comment by xdu-odoo** [comment]
What do yo guys think of this approach ? The problem is that EditableComponent triggers 'o-clicked' event but do not stop propagation of 'click' which is handled by onClick and then close the editable.

**Review by alexkuhn — CHANGES_REQUESTED** [review]
Looks decent. I would mark event in `_onClick` of `EditableText` as handled, and checks whether it was handled by it in `_onClick` of `DiscussSidebarItem`. The advantage of doing this is that it doesn't require ref and it has improved traceability.

File: addons/mail/static/src/components/discuss_sidebar_item/discuss_sidebar_item.js [hunk 1]
```diff
@@ -10,6 +10,7 @@ const useStore = require('mail/static/src/component_hooks/use_store/use_store.js
 const Dialog = require('web.Dialog');
 
 const { Component } = owl;
+const { isEventHandled } = require('mail/static/src/utils/utils.js');
```
Comment (alexkuhn) [inline addons/mail/static/src/components/discuss_sidebar_item/discuss_sidebar_item.js:13]: define just below `const useStore = ...` (and keep single empty line before `const Dialog = ...`)

File: addons/mail/static/src/components/editable_text/editable_text.js [hunk 2]
```diff
@@ -2,6 +2,8 @@ odoo.define('mail/static/src/components/editable_text/editable_text.js', functio
 'use strict';
 
 const { Component } = owl;
+const { markEventHandled } = require('mail/static/src/utils/utils.js');
```
Comment (alexkuhn) [inline addons/mail/static/src/components/editable_text/editable_text.js:5]: define below `'use strict';` with single empty line (and keep single empty line before `const { Component } = ...`)

**Review by alexkuhn — COMMENTED** [review]
Small remarks then good to merge

**Comment by xdu-odoo** [comment]
Ping ?

File: addons/mail/static/src/components/discuss_sidebar_item/discuss_sidebar_item.js [hunk 3]
```diff
@@ -117,6 +118,10 @@ class DiscussSidebarItem extends Component {
      * @param {MouseEvent} ev
      */
     _onClick(ev) {
+        if (isEventHandled(ev, 'EditableText.click')) {
+            // handled in `_onClickedEditableText`
```
Comment (seb-odoo) [inline addons/mail/static/src/components/discuss_sidebar_item/discuss_sidebar_item.js:122]: Not very important but comment is overkill, we know where it comes from with the event handled name.

**Review by seb-odoo — APPROVED** [review]
@robodoo r+

**Pull request closed without merging**